pkg/controllers/scheduler/profile.go

```diff
@@ -30,19 +32,94 @@ import (
 	"github.com/kubewharf/kubeadmiral/pkg/controllers/scheduler/framework/runtime"
 )
 
-func (s *Scheduler) profileForFedObject(_ *unstructured.Unstructured, handle framework.Handle) (framework.Framework, error) {
-	// TODO: generate registry from fed object's type
-	DefaultRegistry := runtime.Registry{
-		apiresources.APIResourcesName:                           apiresources.NewAPIResources,
-		tainttoleration.TaintTolerationName:                     tainttoleration.NewTaintToleration,
-		clusterresources.ClusterResourcesBalancedAllocationName: clusterresources.NewClusterResourcesBalancedAllocation,
-		clusterresources.ClusterResourcesFitName:                clusterresources.NewClusterResourcesFit,
-		clusterresources.ClusterResourcesLeastAllocatedName:     clusterresources.NewClusterResourcesLeastAllocated,
-		rsp.ClusterCapacityWeightName:                           rsp.NewClusterCapacityWeight,
-		placement.PlacementFilterName:                           placement.NewPlacementFilter,
-		clusteraffinity.ClusterAffinityName:                     clusteraffinity.NewClusterAffinity,
-		maxcluster.MaxClusterName:                               maxcluster.NewMaxCluster,
-	}
-
-	return runtime.NewFramework(DefaultRegistry, handle)
+// inTreeRegistry should contain all known in-tree plugins
+var inTreeRegistry = runtime.Registry{
```

Comment:
Should this be versioned?

Reply:
Hmm... that's an interesting suggestion, but i don't see the need to... In any case, there is no versioning of in-tree plugins exposed to the users anyway.

Reply:
What about the enabledByDefault plugins?

Reply:
there's no versioning of that either... i think if we want to version the in-tree plugins or enabledbydefault plugins we can open a new issue to discuss it

Reply:
Didn't we discuss this already - the enabled+disabled way of constructing profiles depends on the default set of enabled plugins being fixed for a specific version - otherwise with the same profile you'll get different resolved plugins - breaking BC.

Reply:
versioning for default plugins is done 😁, as per our discussion, the in-tree registry does not need to be versioned